docs/design/safety/terminology.md

```diff
@@ -0,0 +1,193 @@
+# Safety: Terminology
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Core Terminology](#core-terminology)
+    -   [_Hazard_](#hazard)
+    -   [_Bug_ or _defect_](#bug-or-defect)
+        -   [_Patent_ or _active bug_](#patent-or-active-bug)
+        -   [_Latent bug_](#latent-bug)
+    -   [_Safety_](#safety)
+    -   [_Code_, _software_, or _program safety_](#code-software-or-program-safety)
+    -   [_Safety bugs_](#safety-bugs)
+    -   [_Initial bug_](#initial-bug)
+    -   [_Fail-stop_](#fail-stop)
+-   [Vulnerability Terminology](#vulnerability-terminology)
+    -   [_Vulnerability_ or _security vulnerability_](#vulnerability-or-security-vulnerability)
+    -   [_Vulnerability defense_](#vulnerability-defense)
+        -   [_Detecting_](#detecting)
+        -   [_Mitigating_](#mitigating)
+        -   [_Preventing_ vulnerabilities](#preventing-vulnerabilities)
+        -   [_Ensuring_ correctness](#ensuring-correctness)
+        -   [_Hardening_](#hardening)
+-   [Memory Safety Specifics](#memory-safety-specifics)
+    -   [_Memory safety_](#memory-safety)
+        -   [_Temporal safety_](#temporal-safety)
+        -   [_Spatial safety_](#spatial-safety)
+        -   [_Type safety_](#type-safety)
+        -   [_Initialization safety_](#initialization-safety)
+        -   [_Data-race safety_](#data-race-safety)
+    -   [_Memory safety bug_](#memory-safety-bug)
+    -   [_Memory safe platform_ or _environment_](#memory-safe-platform-or-environment)
+    -   [_Memory safe language_](#memory-safe-language)
+
+<!-- tocstop -->
+
+## Core Terminology
+
+### _Hazard_
+
+Unsafe coding construct that may lead to a bug or vulnerability.
```

Comment:
These definitions aren't really sentences, more like noun phrases. I'm not sure they should have periods.

Reply:
For what it's worth: the ISO drafting rules for standards say that its "Terms and Definitions" entries should be phrases, without leading capitalization or a terminating period, that can be substituted for the term in its context.

I don't think that works so well for the more elaborate definitions you give later (eg, the second sentence in "latent bug" or the qualified definition in "safety"), but I think it would help a lot for the "Vulnerability Terminology" section, where it's currently not completely clear what the link between the terminology and the definition is.